Library/Homebrew/brew.rb

```diff
@@ -65,10 +65,8 @@
   internal_cmd = Commands.valid_internal_cmd?(cmd) || Commands.valid_internal_dev_cmd?(cmd) if cmd
 
   unless internal_cmd
-    require "tap"
-
     # Add contributed commands to PATH before checking.
-    homebrew_path.append(Tap.cmd_directories)
+    homebrew_path.append(Commands.cmd_directories)
```

Comment:
Double command sounds odd. Maybe:

```suggestion
    homebrew_path.append(Commands.tap_directories)
```

to make it clearer this only applies to taps.

Reply:
Yeah, the naming is not great here. Maybe `Commands.tap_cmd_directories`?

Reply:
Sure

Reply:
Addressed in https://github.com/Homebrew/brew/pull/18010/commits/1fe16a5e35952326a97a665ffe9ddde18ee7978b